.pre-commit-config.yaml

```diff
@@ -19,148 +7,27 @@ repos:
       - id: check-merge-conflict # Watch for lingering merge markers.
       - id: check-yaml # Validate all YAML files.
       - id: check-case-conflict # Avoid case sensitivity in file names.
-      - id: debug-statements # Watch for lingering debugger calls.
       - id: end-of-file-fixer # Ensure there's a newline at EOF.
       - id: mixed-line-ending # Only newlines, no line-feeds.
       - id: trailing-whitespace # Remove trailing whitespace.
-      - id: name-tests-test # Follow PyTest naming convention.
 
   ########################################################################################
   # Formatters: hooks that re-write Python and RST files
   ########################################################################################
-
-  # Convert relative imports to absolute imports
-  - repo: https://github.com/MarcoGorelli/absolufy-imports
-    rev: v0.3.1
-    hooks:
-      - id: absolufy-imports
-
-  # Make sure import statements are sorted uniformly.
-  - repo: https://github.com/PyCQA/isort
-    rev: 5.11.4
-    hooks:
-      - id: isort
-
-  # Remove f-string prefix when there's nothing in the string to format.
-  - repo: https://github.com/dannysepler/rm_unneeded_f_str
-    rev: v0.2.0
-    hooks:
-      - id: rm-unneeded-f-str
-
-  # Use built-in types for annotations as per PEP585, with __futures__ for Python 3.8/3.9
-  # Note that this sometimes conflicts with Pydantic on Python 3.8/3.9
-  - repo: https://github.com/sondrelg/pep585-upgrade
-    rev: "v1.0"
-    hooks:
-      - id: upgrade-type-hints
-        args: ["--futures=true"]
-
-  # Update a bunch of Python syntax, using __futures__ for Python 3.8/3.9
-  # Note that this sometimes conflicts with Pydantic on Python 3.8/3.9
-  - repo: https://github.com/asottile/pyupgrade
-    rev: v3.3.1
-    hooks:
-      - id: pyupgrade
-        args: ["--py38-plus"]
-
-  # Deterministic python formatting:
-  - repo: https://github.com/psf/black
-    rev: 22.12.0
-    hooks:
-      - id: black
-        language_version: python3.10
-
   - repo: https://github.com/pre-commit/mirrors-prettier
     rev: v3.0.0-alpha.4
     hooks:
       - id: prettier
-        types_or: [yaml, markdown]
```

Comment:
Don't we want to keep a markdown formatter around for the docs?

Reply:
I removed this because `mdformat` (a different markdown formatter) was being applied in Tox. Then I added `mdformat` as its own pre-commit hook. I've just added the `mdformat-myst` plugin since we're having it format myst markdown.

Reply:
It's hard to see in the diffs though, since the individual lines of the mdformat hook are interleaved with removed lines below.

Reply:
Oh I see. Sounds good to me 👍